inst/scripts/update-metadata-template-dictionaries.R

```diff
@@ -0,0 +1,104 @@
+###############################################
+#### Update Metadata Template Dictionaries ####
+###############################################
+
+if (!require("dccvalidator")) {
+  remotes::install_github("Sage-Bionetworks/dccvalidator")
+}
+library("dccvalidator")
+
+option_list <- list(
+  optparse::make_option(
+    c("-u", "--username"),
+    type = "character",
+    help = "Synapse username or email"
+  ),
+  optparse::make_option(
+    c("-p", "--password"),
+    type = "character",
+    default = NA,
+    help = "Synapse password"
+  ),
+  optparse::make_option(
+    c("-a", "--apikey"),
+    type = "character",
+    default = NA,
+    help = "Synapse apikey"
+  ),
+  optparse::make_option(
+    c("-c", "--config"),
+    type = "character",
+    default = "default",
+    help = "The configuration to use [default = %default]"
+  ),
+  optparse::make_option(
+    c("-d", "--directory"),
+    type = "character",
+    default = ".",
+    help = "output file name [default = %default]"
+  )
+)
+
+opt_parser <- optparse::OptionParser(option_list = option_list);
+opt <- optparse::parse_args(opt_parser);
+
+## Set the configuration to use
+## Change "default" to the appropriate configuration
+Sys.setenv(R_CONFIG_ACTIVE = opt$config)
+
+## Set up Synapse client
+synapse <- reticulate::import("synapseclient")
+syn <- synapse$Synapse()
+
+## Log into Synapse
+if (!is.na(opt$password)) {
+  syn$login(
+    email = opt$username,
+    password = opt$password
+  )
+} else if (!is.na(opt$apikey)) {
+  syn$login(
+    email = opt$username,
+    apiKey = opt$apikey
+  )
+}
+
+## Get annotations
+# Get all annotations
+annots <- purrr::map_dfr(
+  config::get("annotations_table"),
+  get_synapse_annotations,
+  syn = syn
+)
```

Comment:
@karawoo, I thought of this based on our discussion about the two sets of annotation tables potentially having a duplicate. I could add `dplyr::distinct()` here to remove duplicate rows before continuing on; this would stop it from printing the key/values twice. The verification in the next step isn't a problem; it won't fail if there are duplicate rows, only if the columnTypes or descriptions for the same key are not identical.

However, my question is if this _should_ be a separate function that is potentially called in both the shiny app and this script. The function would just return the annotations without duplicated rows. Not sure if a function for this is overkill, though, especially since a cursory look/test with dccvalidator shows that the validator functions should be fine with duplicate rows.